admin/server/reports.go

```diff
@@ -502,6 +553,154 @@ func (s *Server) generateReportName(ctx context.Context, depl *database.Deployme
 	return uuid.New().String(), nil
 }
 
+func (s *Server) issueMagicTokensForExternalEmails(ctx context.Context, orgID, projectID, reportName, ownerID string, emails []string) (map[string]string, error) {
```

Comment:
Since `cleanUpAllReportTokens` does exactly the same as this function does when `len(emails) == 0`, what about removing `cleanUpAllReportTokens` and calling this function with an empty emails slice instead? Could be made clearer by changing the name of this function to `reconcileMagicTokensForExternalEmails` or something like that.